### PR TITLE
Closes #859: bulk upload

### PR DESCRIPTION
**Pull request by cdebourcy**

### CODE REVIEW

File: idseq/__init__.py [hunk 1]
```diff
@@ -138,30 +142,67 @@ def main():
...
+                    args.samplelibrary,
+                    args.samplesequencer,
+                    args.samplenotes,
+                    args.samplememory,
+                    args.host_id,
+                    args.host_genome_name,
+                    args.job_queue)
+            except:
+                print "Failed to upload %s" % sample
+
```
Comment (yunfangjuan) [inline idseq/__init__.py]: cleaner do return here instead of if/else

File: idseq/uploader.py [hunk 2]
```diff
@@ -36,6 +40,30 @@ def parts(self):
...
+    return [f for f in glob.glob(path + wildcards) if re.search(INPUT_REGEX, f) and os.stat(f).st_size > 0]
+
+def detect_samples(path):
+    files_level1 = detect_files(path, level=1)
+    files_level2 = detect_files(path, level=2)
+    if not files_level1 and not files_level2:
+        print "No fastq/fasta files found"
+        raise Exception()
+    samples2files = {}
+    if len(files_level1) >= len(files_level2):
```
Comment (yunfangjuan) [inline idseq/uploader.py]: Hm this is a bit arbitrary?

Comment (yunfangjuan) [reply idseq/uploader.py]: should prolly also document how the files are supposed to be organized? 
we should probably try detect files_level1 first, if we don't find any good pair try level 2.

Comment (cdebourcy) [reply idseq/uploader.py]: Makes sense. But sometimes people have unclean folders, with perhaps a stray fastq file at the top level but then many more actual files in the subfolders?

Comment (yunfangjuan) [reply idseq/uploader.py]: Well. we should ask them to keep things clean there otherwise we will have to deal with many many edge cases.

Comment (cdebourcy) [reply idseq/uploader.py]: ok, fixed

**Pull request merged**